include/BoardData.hxx

```diff
@@ -47,10 +47,20 @@ struct DynaBufferBus : BidirMutexes {
 struct UartBus : DynaBufferBus {};
 struct I2cBus : FixBufferBus<32u> {};
 
+typedef struct  {
+    uint8_t nbr:6;
+    uint8_t isActive:1;
+} ServoPin;
+
+typedef struct {
+    ServoPin Pin;
+    volatile unsigned int angle;
+} servo;
+
 struct BoardData {
     std::vector<std::atomic_bool> digital_pin_values;
     std::vector<std::atomic_uint16_t> analog_pin_values;
-    std::vector<std::atomic_uint8_t> servo_value;
+    std::vector<servo> servo_value;
```

Comment:
You clearly did not care to understand that there is one array element per pin here instead of per-servo; rollback your change